### PR TITLE
[Elastic Agent] Add process error handling guidelines

### PR DESCRIPTION
**Pull request by ruflin**
We don't have a good place yet to document expectations around services run by Elastic Agent. I'm starting this document to have a place to add more content to it but I expect long term we need to figure out a better place.

This guideline comes out of recent issues we had in Cloud and local setups of fleet-server (https://github.com/elastic/fleet-server/pull/883). We never set clear guidlines on what the expectation is of a service run by Elastic Agent and Elastic Agent itself.

This PR is kick off the discussion.

### CODE REVIEW

**Comment by ruflin** [comment]
@elastic/elastic-agent Would be great to get your input on this specific guideline but also on where and how we should document these things.

@graphaelli Thanks for the inspiration to kick this off @simitt might be interested in this too.

File: x-pack/elastic-agent/GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Guidelines
+
+This document contains architecture details around Elastic Agent and guidelines on how new inputs and processes should be built.
+
+## Processes running as service and error handling
+
+All the processes started by Elastic Agent are running as service. Each service is expected to handle local errors on its own and continue working. A process should only fail on startup if a non valid configuration is passed in. As soon as a process is running and partial updates to the config are made without restart, the service is expected to keep running but report the errors.
```
Comment (graphaelli) [inline x-pack/elastic-agent/GUIDELINES.md:7]:
```suggestion
All the processes started by Elastic Agent are running as service. Each service is expected to handle local errors on its own and continue working. A process should only fail on startup if an invalid configuration is passed in. As soon as a process is running and partial updates to the config are made without restart, the service is expected to keep running but report the errors.
```

**Review by graphaelli — COMMENTED** [review]
Would this be a good place to capture security guidelines?  eg configuration of process capabilities (eg via seccomp filters), running as a specific user

**Comment by ruflin** [comment]
@graphaelli Yes. We should start to capture here all the guidelines around Elastic Agent.

**Comment by ruflin** [comment]
@graphaelli Should we get this initial PR merged so we have a future "place" to add more things to the list?

**Comment by graphaelli** [comment]
Sure, that works, thanks for getting this started.

**Review by graphaelli — APPROVED** [review]

**Pull request merged**